2_process/src/match_sites_reaches.R

```diff
@@ -0,0 +1,49 @@
+#' original author: David Watkins
+#' from https://code.usgs.gov/wwatkins/national-site-reach/-/blob/master/R/match_sites_reaches.R
+#' modified by: Jeff Sadler
+#' Match each site with a reach (seg_id/COMID)
+get_site_flowlines <- function(reach_sf, sites) {
+  #' 
+  #' @description Function to match reaches (flowlines) to point sites 
+  #'
+  #' @param reach_sf sf object of reach polylines with column "segidnat" and in WGS84
+  #' @param sites dataframe with columns "lat" "lon"
+  #'
+  #' @value A data frame the same columns as the sites input dataframe with additional columns
+  #' of "segidnat" and "offset" where "offset" is the distance (in degrees) between the point
+  #' and matching flowline
+  
+  reaches_nhd_fields <- reach_sf %>%
+    select(COMID = segidnat) %>%
+    mutate(REACHCODE = COMID, ToMeas = 100, FromMeas = 100) %>%
```

Comment:
To be honest I do not know what these two arguments are. I too was unclear looking at the `nhdPlusTools::get_flowline_index` documentation

Reply:
`ToMeas` and `FromMeas` are attributes within the NHD that describe distances along the reach codes. These arguments are required inputs to `get_flowline_index`, which (I think) just uses them to estimate the proportion along the reach where the point gets 'snapped' to the matched flowline. My guess is that David sets both values to 100 here so that `get_flowline_index` will run. The consequence is that if we interpreted the full output of `get_flowline_index` (its looks like David only saves a subset of the variables), all pts would be snapped to the downstream end of the matched flowline.

Reply:
We might also consider adding a comment here (as a reminder to ourselves) that notes the `COMID` is the reach variable over which the site points will be matched to. `segidnat` is not actually equivalent to the associated NHDPlus flowline COMID. Similarly, `ToMeas` and `FromMeas` should not be interpreted using their NHDPlus meaning and instead are acting as placeholders used to apply `get_flowline_index` to the PRMS network.

Reply:
good idea.